app/controllers/appointments_controller.rb

```diff
@@ -66,11 +66,18 @@ def search
       search_params[:q],
       search_params[:date_range]
     )
+
+    check_for_exact_match
```

Comment:
I'd shuffle this around a bit. Firstly `redirect_on_exact_match` makes more sense as it's a command not a query. Secondly I'd pass `@search.results` as an argument to this method and move the cardinality check up to here, that was we can return early, otherwise the code below still runs:

```ruby
return redirect_on_exact_match(@search.results) if @search.results.count.one?

def redirect_on_exact_match(results)
  redirect_to #...
end
```